### PR TITLE
Dim reduction plugin, #43. #44

### PR DESCRIPTION
**Pull request by mrossNSTG**
Various improvements/implementations of input validation and clarifications in script.

- Check that filetype is from valid list, or FAIL
- Allow NULL *or* NA for color_by
- Linked reference to vignette for color and RBrewer palette options
- Amendment of extend_palette to avoid warning when extending < 3 colors
- Set up palette in the case of plot_colors = NULL or NA
- Test for valid colors, with override if first value of list is an RBrewer palette

### CODE REVIEW

**Review by Maggieer01 — COMMENTED** [review]
I think **color_levels** fails to catch invalid names, I tested it with invalid names such as _c("dog", "cat")_ and _c("dsfasf","sfsdfa")_, but it does not affect anything and there is no error message.

File: DimReduction/DimReduction.R [hunk 1]
```diff
@@ -112,49 +118,67 @@ main <- function(dataset, segmentAnnotations, targetAnnotations, outputFolder) {
...
+  }
+  
+  # if color is a gene symbol add it to the annotations for plotting, catch NA or NULL
+  if(is.null(color_by) | is.na(color_by)) {
     colType <- 'Null'
   } else if(!color_by %in% rownames(targetCountMatrix) &
             !color_by %in% colnames(segmentAnnotations)) {
     fail(message = 'Color not found. Please confirm that your color feature is either a column in Segment Properties or a Target name from your target count matrix')
-  } else if(color_by %in% rownames(targetCountMatrix)) {
+  } else if(!color_by %in% rownames(targetCountMatrix) &
```
[the remark this answers is not in the document]
Comment (mrossNSTG) [reply DimReduction/DimReduction.R:132]: Hmm, yeah, looks like I made a hash of that logic. This is the case of color by a single target, so really the only logic gate should be:

  } else if ( color_by %in% rownames(targetCountMatrix ) {
 
...which, come to think of it, it should be explicitly *one* target, not more. I should fix that as well.

**Review by Maggieer01 — APPROVED** [review]
Hi Marty, good job, all files look good! 

I have reviewed the code and tested the the plugin works, issues listed in #43 and #44 are fixed. The vignette, changelog and version are appropriately updated.

**Comment by mrossNSTG** [comment]
A late request from CX to set seed on UMAP to ensure reproducible results. Added to Reqs (#8) as well.

**Comment by NicoleEO** [comment]
This has been merged into dev already. Dev to be merged into master later.

**Pull request closed without merging**